### PR TITLE
Update the stereo sample:

### PR DESCRIPTION
**Pull request by pemmanuelviel**
* add the HH4 mode
* option to display disparity with a color map
* display current settings in the title bar
* don't close app when wanting to take screenshots

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/cpp/stereo_match.cpp [hunk 1]
```diff
@@ -297,5 +290,41 @@ int main(int argc, char** argv)
...
+        oss << "  blocksize:" << (alg==STEREO_BM ? SADWindowSize : sgbmWinSize);
+        oss << "  max-disparity:" << numberOfDisparities;
+        disp_name += oss.str();
+
+        namedWindow("left", cv::WINDOW_NORMAL);
+        imshow("left", img1);
+        namedWindow("right", cv::WINDOW_NORMAL);
+        imshow("right", img2);
+        namedWindow(disp_name, cv::WINDOW_AUTOSIZE);
+        if (color_display) {
```
Comment (mshabunin) [inline samples/cpp/stereo_match.cpp:312]: Please use consistent formatting across the file:
```.cpp
if (...)
{
    ...
}
```

File: samples/cpp/stereo_match.cpp [hunk 2]
```diff
@@ -268,19 +274,6 @@ int main(int argc, char** argv)
...
-        namedWindow("disparity", 0);
-        imshow("disparity", disp8);
-        printf("press any key to continue...");
-        fflush(stdout);
-        waitKey();
-        printf("\n");
-    }
 
     if(!disparity_filename.empty())
         imwrite(disparity_filename, disp8);
```
Comment (asmorkalov) [inline samples/cpp/stereo_match.cpp:279]: `--color` option handling is missed here.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

**Review by mshabunin — APPROVED** [review]

**Comment by pemmanuelviel** [comment]
It's on purpose.
Datasets are giving ground truth on one channel (8 or more bits that can be
converted to 8 bits). Storing the image on one channel also allows direct
comparison with these ground truths, knowing in addition that the choice of
the color map might not be the same one from one dataset to another one
when used for display.
If we want to store color mapped disparity images for visual comparison,
screenshot is an option as it keeps track in the title bar of the different
settings used for the process.
Are you fine with that?

**Comment by asmorkalov** [comment]
The default behavior is to produce one channel disparity image. It works with datasets as you expect. I just propose to have the consistent option behavior for case with window and in "no-display" mode. It can be used for testing.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**